XML Schema test

Yurasb opened the pull request:
1. Chages to url
2. Adding test for XML validation with XSD schema
3. Adding XSD schema for valid case

Yurasb committed:
1. Chages to url
2. Adding test for XML validation with XSD schema
3. Adding XSD schema for valid case

Yurasb committed:
Small optimization - removing StringIO from schema

v1-wizard commented on test_cases/test_cases.py:
```diff
@@ -1,121 +1,31 @@
 # -*- coding: utf-8 -*-
```
Empty line between utf-8 comment and imports.

v1-wizard commented on test_cases/test_cases.py:
```diff
@@ -1,121 +1,31 @@
-         '</script>\n'
-         '</body>\n'
-         '</html>\n'
+    xsd=open('../test_cases/valid_case.xsd', 'r')
```
Could you change xsd as file to xsd_path and hide in case object open(...)?

Yurasb replied:
Something like that?

`class Case(object):
    def __init__self, xsd_path):
        self.xsd = open(xsd_path, 'r')
........
    xsd_path='../test_cases/valid_case.xsd'`

v1-wizard replied:
Yes. Also you could read about with statement. https://docs.python.org/3/reference/compound_stmts.html#the-with-statement https://docs.python.org/2/tutorial/inputoutput.html

> It is good practice to use the with keyword when dealing with file objects. This has the advantage that the file is properly closed after its suite finishes, even if an exception is raised on the way. It is also much shorter than writing equivalent try-finally blocks:

v1-wizard commented on tests/test_headers.py:
```diff
@@ -1,55 +1,40 @@
-    assert 'Expires' in headers
-
-
 def test_content_encoding(headers, case):
```
Tricky. Use case as dependency to response and response as dependency to headers and use case as single fixture. https://www.python.org/dev/peps/pep-0020/#id3 - "Explicit is better than implicit."

v1-wizard reviewed (changes requested):
Let's talk about? =)

Yurasb committed:
Fixing typo

Yurasb committed:
Open xsd scheme in class constructor

Yurasb committed:
1. using only one fixture - get_case, response and related to it (status_code, headers) moved to tests as steps
2. util methods for xmlschema assertion - get_xmlschema for case and get_doc for test
3. getting rid of shdowing names
4. rename test_cases.py to cases.py

Yurasb committed:
Change to URL creation - now with format

Yurasb committed:
Changing name of property

Yurasb committed:
1. adding valid case without objects
2. parametrizing case fixture

Yurasb committed:
Rename

v1-wizard commented on main.py:
```diff
@@ -1,15 +1,11 @@
-      ':' +\
-      configuration['port'] +\
-      '/#map='
+url = '{}://{}:{}/api/{}/map?bbox='.format(
```
You could do it like a {protocol}://{host}:{port}/api/{api_version} but this is fine too.

v1-wizard replied:
.format(**kwarg) if in configuration only 4 your keys.

v1-wizard commented on test_cases/cases.py:
```diff
@@ -0,0 +1,58 @@
+               u'31.17742896080017,52.56634718820995',
+    status_code=200,
+    headers={
+        'Cache-Control': 'private, max-age=0, must-revalidate',
```
Do you really need all this headers for checking?

v1-wizard replied:
Let's talk about.

v1-wizard commented on tests/__init__.py:
```diff
@@ -0,0 +1,9 @@
+# -*- coding: utf-8 -*-
```
blank line.

v1-wizard commented on tests/test_headers.py:
```diff
@@ -1,55 +1,47 @@
 # -*- coding: utf-8 -*-
+import requests
```
blank line

v1-wizard commented on tests/conftest.py:
```diff
@@ -1,34 +1,9 @@
-def case(request):
-    case = valid_case
+@pytest.fixture(scope='function', params=range(2))
+def get_case(request):
```
Maybe context?
